mitodl/open-discussions
# Fields for determining OCW URLs

Depends on #3141

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #3144 

#### What's this PR do?
Adds additional fields to the LearningResourceRun & ContentFile models and ES index that can be used to construct OCW URL's.

#### How should this be manually tested?
- Recreate the index
- Run `backpopulate_ocw_files` (assuming you already imported OCW courses, if not run `backpopulate_ocw_data --overwrite --skip-s3`)
- Check that the ES index contains fields for `department_slug` under each run.
- Run this search via curl/postman for resources.  Each resource should include a `section_slug` field (may be null), and a `run_department_slug` field.

```
{  "size": 100,
    "post_filter": {
        "bool": {
            "must": [
                {
                    "bool": {
                        "should": [
                            {
                                "term": {
                                    "object_type.keyword": "resourcefile"
                                }
                            }
                        ]
                    }
                },
                {
                    "bool": {
                        "should": [
                            {
                                "has_parent": {
                                    "parent_type": "resource",
                                    "query": {
                                        "bool": {
                                            "should": [
                                                {
                                                    "term": {
                                                        "offered_by": "OCW"
                                                    }
                                                }
                                            ]
                                        }
                                    }
                                }
                            }
                        ]
                    }
                }
            ]
        }
    }
}
```
